Neos.ContentRepository.Core/Classes/Projection/ContentGraph/Node.php

```diff
@@ -50,15 +50,22 @@ public function __construct(
          * With the above information, you can fetch a Subgraph using {@see ContentGraphInterface::getSubgraph()}.
          * or {@see \Neos\ContentRepositoryRegistry\ContentRepositoryRegistry::subgraphForNode()}
          */
-        public readonly NodeAggregateId $nodeAggregateId,
+        public NodeAggregateId $nodeAggregateId,
         /**
          * returns the DimensionSpacePoint the node is at home in. Usually needed to address a Node in a NodeAggregate
          * in order to update it.
          */
-        public readonly OriginDimensionSpacePoint $originDimensionSpacePoint,
-        public readonly NodeAggregateClassification $classification,
-        public readonly NodeTypeName $nodeTypeName,
-        public readonly NodeType $nodeType,
+        public OriginDimensionSpacePoint $originDimensionSpacePoint,
+        public NodeAggregateClassification $classification,
+        /**
+         * The node's node type name; always set, even if unknown to the NodeTypeManager
+         */
+        public NodeTypeName $nodeTypeName,
+        /**
+         * The node's node type, null if unknown to the NodeTypeManager
+         * @deprecated Ask the NodeTypeManager instead
+         */
+        public ?NodeType $nodeType,
```

Comment:
+1 as we discussed its now nullable and deprecated!

Reply:
We should adjust the deprecation notice in that we probably want to replace the $nodeType object and not remove the property: eg the concept of this big NodeType god object directly on the node is deprecated, but there will always be some kind of schema information - even if less powerful - on the node

Reply:
documentation has been adjusted